Find a functionally equivalent code:android.app.Notification.cloneInto:COMMENT
Method Modifier: public      hidden      
Comment:/**
 * Copy all (or if heavy is false, all except Bitmaps and RemoteViews) members
 * of this into that.
 * @hide
 */

@@ -22,27 +22,30 @@
     // android.net.Uri is immutable
     that.sound = this.sound;
     that.audioStreamType = this.audioStreamType;
+    if (this.audioAttributes != null) {
+        that.audioAttributes = new AudioAttributes.Builder(this.audioAttributes).build();
+    }
     final long[] vibrate = this.vibrate;
     if (vibrate != null) {
         final int N = vibrate.length;
@@ -34,45 +37,45 @@
     that.defaults = this.defaults;
     that.flags = this.flags;
     that.priority = this.priority;
-    final String[] thiskind = this.kind;
-    if (thiskind != null) {
-        final int N = thiskind.length;
-        final String[] thatkind = that.kind = new String[N];
-        System.arraycopy(thiskind, 0, thatkind, 0, N);
-    }
+    that.category = this.category;
+    that.mGroupKey = this.mGroupKey;
+    that.mSortKey = this.mSortKey;
     if (this.extras != null) {
         try {
             that.extras = new Bundle(this.extras);
@@ -59,64 +59,73 @@
     if (heavy && this.bigContentView != null) {
         that.bigContentView = this.bigContentView.clone();
     }
+    if (heavy && this.headsUpContentView != null) {
+        that.headsUpContentView = this.headsUpContentView.clone();
+    }
+    that.visibility = this.visibility;
+    if (this.publicVersion != null) {
+        that.publicVersion = new Notification();
+        this.publicVersion.cloneInto(that.publicVersion, heavy);
+    }
+    that.color = this.color;
     if (!heavy) {
         // will clean out extras
         that.lightenPayload();

